File: content/code-examples/tested/java/driver-sync/timeseries/SecondaryIndexes.snippet.sort-and-group.java
var pipeline = Arrays.asList(
        new Document("$sort", new Document()
                .append("metadata.sensorId", 1)
                .append("timestamp", -1)),
        new Document("$group", new Document()
                .append("_id", "$metadata.sensorId")
                .append("ts", new Document("$first", "$timestamp"))
                .append("temperatureF", new Document("$first", "$currentConditions.tempF")))
);

List<Document> result = collection.aggregate(pipeline).into(new ArrayList<>());

File: content/code-examples/tested/python/pymongo/timeseries/secondary_indexes.snippet.sort-and-group.py
pipeline = [
    {"$sort": {"metadata.sensorId": 1, "timestamp": -1}},
    {"$group": {
        "_id": "$metadata.sensorId",
        "ts": {"$first": "$timestamp"},
        "temperatureF": {"$first": "$currentConditions.tempF"}
    }}
]

result = list(collection.aggregate(pipeline))
